topics/introduction/tutorials/data-manipulation-olympics/tutorial.md

```diff
@@ -570,6 +571,135 @@ This file contains a lot of data, but we may only be interested in a subset of t
+>
+> - **Boolean Expressions**: These are expressions that evaluate to either `True` or `False`. For example, `c2.startswith('Liu')` will return `True` if the value in column 2 starts with 'Liu', and `False` otherwise.
+>
+> - **Comparison Operators**: These include `==` (equal to), `!=` (not equal to), `<` (less than), `>` (greater than), `<=` (less than or equal to), and `>=` (greater than or equal to). These operators are used to compare values and are commonly used in filtering.
+>
+> - **String Methods**: Strings in Python have several built-in methods that can be used for filtering. Some commonly used ones include:
+>   - `startswith(substring)`: Checks if a string starts with a specified substring.
+>   - `endswith(substring)`: Checks if a string ends with a specified substring.
+>   - `find(substring)`: Returns the lowest index of the substring if it is found in the string; otherwise, it returns `-1`.
+>   - `count(substring)`: Returns the number of non-overlapping occurrences of the substring in the string.
+>
+{: .comment}
+
+These concepts will be used in the filtering expressions we write in Galaxy. Let's now explore some specific scenarios where you might use these expressions.
+
+
+> <hands-on-title>Filtering Based on Substring Occurrence</hands-on-title>
+>
+>    We want to filter rows where a particular column contains a specific substring using the `cX.find()` method, where `X` is the column number.
+>
+>    For example, while you can easily filter Summer or Winter Olympics using `c13=="Summer"` or `c13=="Winter"` if the data is stored in a dedicated column, sometimes the data may not be that straightforward. The relevant information might be embedded in another column, like in `c11` with entries such as "1992 Summer Olympics".
+>
+>    In such cases, you can use substring filtering:
+>
+>    > <question-title></question-title>
+>    >
+>    > 1. How would you filter the rows where the 11th column contains "Summer"?
+>    > 2. How would you filter the rows where the 11th column does **not** contain "Summer"?
+>    >
+>    > > <solution-title>Answers</solution-title>
+>    > >
+>    > > 1. `c11.find('Summer') != -1`
+>    > > 2. `c11.find('Summer') == -1`
+>    > >
+>    > {: .solution}
+>    {: .question}
+>
+>    Ok, great, now that you've got the hang of writing expressions for this tool, let's filter the file for rows where column 11 contains "Summer":
+>
+> 1. {% tool [**Filter** data on any column using simple expressions]({{version_filter}}) %} with the following parameters:
+>    - {% icon param-file %} *"Filter"*: `olympics.tsv`
+>    - {% icon param-text %} *"With the following condition"*: `c11.find('Summer') != -1`
+>    - {% icon param-text %} *"Number of header lines to skip"*: `1`
+>
+> 2. {% icon galaxy-eye %} **View** the filtered file.
+>
+> 3. **Rename** {% icon galaxy-pencil %} both outputs to something descriptive.
+>
```

Comment:
could you add a small Q&A box at the end of each hands-on box where people can check their answers? So here it would just be?

Q: How many rows contained the string "Summer" in column 11?
A: <number of columns>

Reply:
I've added the small Q&A box at the end of each hands-on box as you suggested. Thank you for the great idea!

Reply:
Thanks!